src/renderers/shared/ReactDOMFrameScheduling.js

```diff
@@ -23,10 +23,33 @@
 import type {Deadline} from 'ReactFiberReconciler';
 
 var invariant = require('fbjs/lib/invariant');
+var ExecutionEnvironment = require('fbjs/lib/ExecutionEnvironment');
 
 // TODO: There's no way to cancel these, because Fiber doesn't atm.
 let rAF: (callback: (time: number) => void) => number;
 let rIC: (callback: (deadline: Deadline) => void) => number;
+
+if (!ExecutionEnvironment.canUseDOM) {
+  global.requestAnimationFrame = function(
```

Comment:
But that wont prevent the codepath accessing `requestAnimationFrame` (and turns out `requestIdleCallback` too. [L101](https://github.com/prometheansacrifice/react/blob/e99eb0aa133852ad4153dd58973689b880a4966b/src/renderers/shared/ReactDOMFrameScheduling.js#L101) is breaking on the server too.

So wrap the entire polyfilling code in else block of the `if (!ExecutionEnvironment.canUseDOM) {`?